inc/classes/subscriber/CDN/RocketCDN/NoticesSubscriber.php

```diff
@@ -0,0 +1,265 @@
+	/**
+	 * RocketCDN API Client instance.
+	 *
+	 * @var APIClient
+	 */
+	private $api_client;
+
+	/**
+	 * Constructor
+	 *
+	 * @param APIClient $api_client RocketCDN API Client instance.
+	 * @param string    $template_path Path to the templates.
+	 */
+	public function __construct( APIClient $api_client, $template_path ) {
+		parent::__construct( $template_path );
+
+		$this->api_client = $api_client;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'admin_notices'                    => [
+				[ 'promote_rocketcdn_notice' ],
+				[ 'purge_cache_notice' ],
+			],
+			'rocket_before_cdn_sections'       => 'display_rocketcdn_cta',
+			'wp_ajax_toggle_rocketcdn_cta'     => 'toggle_cta',
+			'wp_ajax_rocketcdn_dismiss_notice' => 'dismiss_notice',
+			'admin_footer'                     => 'add_dismiss_script',
+		];
+	}
+
+	/**
+	 * Adds notice to promote Rocket CDN on settings page
+	 *
+	 * @since 3.5
+	 * @author Remy Perona
+	 *
+	 * @return void
+	 */
+	public function promote_rocketcdn_notice() {
+		if ( ! $this->should_display_notice() ) {
+			return;
+		}
+
+		echo $this->generate( 'promote-notice' );
+	}
+
+	/**
+	 * Adds inline script to permanently dismissing the RocketCDN promotion notice
+	 *
+	 * @since 3.5
+	 * @author Remy Perona
+	 *
+	 * @return void
+	 */
+	public function add_dismiss_script() {
+		if ( ! $this->should_display_notice() ) {
+			return;
+		}
+
+		$nonce = wp_create_nonce( 'rocketcdn_dismiss_notice' );
+		?>
+		<script>
+		window.addEventListener( 'load', function() {
+			var dismissBtn  = document.querySelectorAll( '#rocketcdn-promote-notice .notice-dismiss, #rocketcdn-promote-notice #rocketcdn-learn-more-dismiss' );
+
+			dismissBtn.forEach(function(element) {
+				element.addEventListener( 'click', function( event ) {
+					var httpRequest = new XMLHttpRequest(),
+						postData    = '';
+
+					postData += 'action=rocketcdn_dismiss_notice';
+					postData += '&nonce=<?php echo esc_attr( $nonce ); ?>';
+					httpRequest.open( 'POST', '<?php echo esc_url( admin_url( 'admin-ajax.php' ) ); ?>' );
+					httpRequest.setRequestHeader( 'Content-Type', 'application/x-www-form-urlencoded' )
+					httpRequest.send( postData );
+				});
+			});
+		});
+		</script>
+		<?php
+	}
+
+	/**
+	 * Checks if the promotion notice should be displayed
+	 *
+	 * @since 3.5
+	 * @author Remy Perona
+	 *
+	 * @return boolean
+	 */
+	private function should_display_notice() {
+		if ( ! current_user_can( 'rocket_manage_options' ) ) {
+			return false;
+		}
+
+		if ( 'settings_page_wprocket' !== get_current_screen()->id ) {
+			return false;
+		}
+
+		if ( get_user_meta( get_current_user_id(), 'rocketcdn_dismiss_notice', true ) ) {
+			return false;
+		}
+
+		$subscription_data = $this->api_client->get_subscription_data();
+
+		if ( $subscription_data['is_active'] ) {
+			return false;
+		}
+
+		return true;
```

Comment:
Return the inverse of the conditional expression instead to reduce the code here.

Reply:
Resolved a20d0d22ca178b296f4555109538770940915ce8 and 2bde1776b1a6e1963960bbb89b5f7632cc1b0382